File: mine/src/main/java/com/leory/badminton/mine/mvp/ui/activity/SettingActivity.java
package com.leory.badminton.mine.mvp.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.leory.badminton.mine.R;
import com.leory.badminton.mine.R2;
import com.leory.badminton.mine.mvp.model.sp.AccountSp;
import com.leory.commonlib.base.BaseActivity;
import com.leory.commonlib.widget.XSDToolbar;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Describe : 设置页面
 * Author : leory
 * Date : 2019-06-20
 */
public class SettingActivity extends BaseActivity {
    @BindView(R2.id.toolbar)
    XSDToolbar toolbar;

    public static void launch(Activity preActivity) {
        preActivity.startActivity(new Intent(preActivity, SettingActivity.class));
    }

    @Override
    public int initView(@Nullable Bundle savedInstanceState) {
        return R.layout.activity_setting;
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        toolbar.setOnBackListener(new XSDToolbar.OnBackListener() {
            @Override
            public void onBackClick() {
                finish();
            }
        });
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // TODO: add setContentView(...) invocation
        ButterKnife.bind(this);
    }

    @OnClick({R2.id.btn_logout})
    public void onViewClicked(View view) {
        if (view.getId() == R.id.btn_logout) {
            AccountSp.putLoginState(false);
            finish();
        }
    }
}

File: mine/src/main/java/com/leory/badminton/mine/mvp/ui/activity/SettingActivity.kt
package com.leory.badminton.mine.mvp.ui.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.leory.badminton.mine.R
import com.leory.badminton.mine.mvp.model.sp.AccountSp
import com.leory.commonlib.base.BaseActivity
import com.leory.commonlib.mvp.IPresenter
import kotlinx.android.synthetic.main.activity_setting.*

/**
 * Describe : 设置页面
 * Author : leory
 * Date : 2019-06-20
 */
class SettingActivity : BaseActivity<IPresenter>() {

    override fun initView(savedInstanceState: Bundle?): Int = R.layout.activity_setting


    override fun initData(savedInstanceState: Bundle?) {
        toolbar.setOnBackListener { finish() }
        btn_logout.setOnClickListener { onViewClicked(it) }
    }


    private fun onViewClicked(view: View) {
        if (view.id == R.id.btn_logout) {
            AccountSp.loginState = false
            finish()
        }
    }

    companion object {

        @JvmStatic
        fun launch(preActivity: Activity) {
            preActivity.startActivity(Intent(preActivity, SettingActivity::class.java))
        }
    }
}
